test/integration/render-tests/text-field/property-function/style.json

```diff
@@ -0,0 +1,74 @@
+{
+  "version": 8,
+  "metadata": {
+    "test": {
+      "height": 256
+    }
+  },
+  "center": [
+    13.418056,
+    52.499167
+  ],
+  "zoom": 14,
+  "sources": {
+    "point": {
+      "type": "geojson",
+      "data": {
+        "type": "FeatureCollection",
+        "features": [
+          {
+            "type": "Feature",
+            "properties": { "x": 1, "units": "furlongs" },
+            "geometry": {
+              "type": "Point",
+              "coordinates": [
+                13.413056,
+                52.499167
+              ]
+            }
+          },
+          {
+            "type": "Feature",
+            "properties": { "x": 0, "units": "parsecs" },
+            "geometry": {
+              "type": "Point",
+              "coordinates": [
+                13.423056,
+                52.499167
+              ]
+            }
+          }
+        ]
+      }
+    }
+  },
+  "glyphs": "local://glyphs/{fontstack}/{range}.pbf",
+  "layers": [
+    {
+      "id": "background",
+      "type": "background",
+      "paint": {
+        "background-color": "white"
+      }
+    },
+    {
+      "id": "text",
+      "type": "symbol",
+      "source": "point",
+      "layout": {
+        "text-field": {
+          "type": "categorical",
+          "property": "x",
+          "stops": [
+            [0, "Zero {units}"],
```

Comment:
I am not sure we want to allow the combination of data-driven properties and interpolation syntax. It could further complicate the future of features like https://github.com/mapbox/mapbox-gl-style-spec/issues/47 and https://github.com/mapbox/mapbox-gl-style-spec/issues/104.